src/components/chatX/ChatXPage.js

```diff
@@ -0,0 +1,118 @@
+import React from 'react';
+import { Link } from 'react-router';
+import checkAuth from '../requireAuth';
+import ChatXBody from './ChatXBody';
+import ChatXInput from './ChatXInput';
+import { messageSent, getInitial10Messages, getNewMessages } from '../../actions/chatXActions';
+import { bindActionCreators } from 'redux';
+import { connect } from 'react-redux';
+
+export class ChatXPage extends React.Component {
+    constructor(props, context) {
+        super(props, context);
+        this.user = this.props.user.email;
+
+        this.state = {
+            newMessage: ""
+        };
+
+        this.updateInputState = this.updateInputState.bind(this);
+        this.sendMessage = this.sendMessage.bind(this);
+        this.setScrollToBottom = this.setScrollToBottom.bind(this);
+    }
+
+    componentDidMount() {
+        this.props.actions.getNewMessages();
+        this.props.actions.getInitial10Messages();
+    }
+
+    componentDidUpdate(prevProps, prevState) {
+        if (this.isMessageListDifferent(prevProps.chatX.msgList, this.props.chatX.msgList)) {
+            this.setScrollToBottom();
+        }
+    }
+
+    isMessageListDifferent(prev, current) {
+        const messageProperties = [
+            'content',
+            'sender',
+            'isSelf'
+        ];
+        if (prev.length !== current.length) {
+            return true;
+        }
+
+        Object.keys(current).map(key => {
+            messageProperties.map((prop) => {
+                if (prop in current[key]) {
+                    if (current[key][prop] !== prev[key][prop]) {
+                        return true;
+                    }
+                }
+            });
+        });
+        return false;
+    }
+
+    updateInputState(messageContent) {
+        return this.setState({ newMessage: messageContent });
+    }
+
+    sendMessage() {
+        if (this.state.newMessage) {
+            this.props.actions.messageSent({
+                content: this.state.newMessage,
+                sender: this.user
+            })
+                .then(() => {
+                    this.setState({ 
+                        newMessage: "" 
+                    });
+                });
+        }
+    }
+
+    setScrollToBottom() {
+        let el = document.getElementById("chat-x-body");
```

Comment:
http://jamesknelson.com/react-js-by-example-interacting-with-the-dom/

Reply:
Since ChatXBody is a stateless function component, should I put a ref in a div wrapping ChatXBody in ChatXPage render function? Or convert ChatXBody into a class?